### PR TITLE
Property test for MArray.slice()

### PR DESCRIPTION
**Pull request by justmv**
Implements a property test for MArray.slice()

- Added fast-check library
- Added `/tests/marray.test.ts`
- Fixed `slice()` in `MArray` for cases where the `at < 0` & `at > arr.length`

### CODE REVIEW

File: tests/marray.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            const delsArray = data.splice(record.at, record.deleteCount, ...record.add);
+            const delsMArray = mArray.splice(record.at, record.deleteCount, ...record.add);
+
+            expect(delsMArray.toArray()).toEqual(delsArray);
+            expect(mArray.toArray()).toEqual(data);
+          }
+        },
+      ),
+      {
+        numRuns: 50,
```
Comment (spion) [inline tests/marray.test.ts:31]: Does the default value (100) take too long?

File: tests/marray.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import * as fc from 'fast-check';
```
Comment (spion) [inline tests/marray.test.ts:1]: Hmm. Why place the file in `tests` not next to the source like the other specs?

**Comment by spion** [comment]
I think this is good to go, pretty much!

File: tests/marray.test.ts (hunk 2 above)
Comment (justmv) [reply tests/marray.test.ts]: I saw that the tests for the tree were there so I put it there, moved it to `src` for `MArray` consistency.

Commit: [98a1271](https://github.com/hfour/yads/pull/33/commits/98a127190020004645a81f370e7473c081427f41)

File: tests/marray.test.ts (hunk 1 above)
Comment (justmv) [reply tests/marray.test.ts]: Turns out it doesn't -- only ~10 sec longer 😄 

Commit: [252c15a](https://github.com/hfour/yads/pull/33/commits/252c15a201fe14a9952deba2d1aaedf946c962f2)

**Pull request merged**